flipper-plugin-realm/src/components/objectManipulation/types/ObjectInput.tsx

```diff
@@ -18,30 +19,36 @@ export const ObjectInput = ({
     instance.state
   );
 
-  const [value, setValue] = useState<IndexableRealmObject>(defaultValue as IndexableRealmObject);
-  const [chosen, setChosen] = useState(!!value);
+  const [serializedObject, setSerializedObject] = useState<DeserializedRealmObject>(defaultValue as DeserializedRealmObject);
+  const [chosen, setChosen] = useState(!!serializedObject);
   const [visible, setVisible] = useState(false);
-  const [objects, setObjects] = useState<IndexableRealmObject[]>([]);
+  const [objects, setObjects] = useState<DeserializedRealmObject[]>([]);
   const [hasMore, setHasMore] = useState(false);
   const [cursor, setCursor] = useState<string | null>(null);
   const [totalObjects, setTotalObjects] = useState(0);
 
   const targetSchema = schemas.find(
     (schema) => schema.name === property.objectType
   );
+  const isEmbedded = targetSchema?.embedded;
+
 
   if (!targetSchema) {
     return <>Target schema {property.objectType} not found</>;
   }
 
   const renderChosen = () => {
     let content;
-    if (targetSchema?.primaryKey !== undefined) {
-      const val = value[targetSchema.primaryKey];
-      content = `${targetSchema.primaryKey}: ${val}`;
+    if(isEmbedded) {
+      content = ``;
     }
+    // TODO: Fetch object with getObject to display primary key instead of objectKey
```

Comment:
Did you want to address this in a separate PR, or is this doable now?

Reply:
I was thinking of separate PR. although I just thought of a way to make this simpler, i.e. just send over primaryKey if it exists along with objectType / objectKey for references, so I might just do this. I generally avoid leaving TODOs like this but because of the state of the code right now this seemed like the quickest way to leave some bits for future planning.